transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/specification/ws/connector/BaseFramingIT.java

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright 2014, Kaazing Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.gateway.transport.wsn.specification.ws.connector;
+
+import static org.junit.Assert.assertTrue;
+import static org.kaazing.test.util.ITUtil.createRuleChain;
+
+import java.nio.ByteBuffer;
+import java.util.Random;
+
+import org.apache.mina.core.filterchain.IoFilterChain;
+import org.apache.mina.core.future.ConnectFuture;
+import org.apache.mina.core.service.IoHandler;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+import org.kaazing.gateway.transport.IoHandlerAdapter;
+import org.kaazing.gateway.transport.ws.bridge.filter.WsBuffer;
+import org.kaazing.gateway.transport.wsn.WsnProtocol;
+import org.kaazing.gateway.transport.wsn.WsnSession;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+import org.kaazing.mina.core.buffer.IoBufferAllocatorEx;
+import org.kaazing.mina.core.buffer.IoBufferEx;
+import org.kaazing.mina.core.session.IoSessionEx;
+
+public class BaseFramingIT {
+    private final K3poRule k3po = new K3poRule().setScriptRoot("org/kaazing/specification/ws/framing");
+
+    private final WsnConnectorRule connectorRule = new WsnConnectorRule();
+
+    @Rule
+    public TestRule chain = createRuleChain(connectorRule, k3po);
+
+    @Test
+    @Ignore("Exception: message is empty. Forgot to call flip()?")
+    @Specification({
+        "echo.binary.payload.length.0/handshake.response.and.frame"
+        })
+    public void shouldEchoBinaryFrameWithPayloadLength0() throws Exception {
+        IoHandler handler = newIoHandlerForBaseFraming(WsBuffer.Kind.BINARY, 0);
```

Comment:
Recommend using jmock instead of an `IoHandler` implementation?
Recommend inlining the mocked behavior to make each test readable and independent, otherwise deciphering each test requires reverse engineering the multi-purpose handler implementation.